src/main.cpp

```diff
@@ -212,15 +216,22 @@ int main(int argc, char** argv) {
     HttpClientConfig config = {elastic_addr, sslEnabled, caPath, username,
                                password};
     if (reindex) {
-      LOG_INFO("Create Elasticsearch index at " << elastic_index);
-      Reindexer *reindexer = new Reindexer(connection_string.c_str(),
-                                           database_name.c_str(),
-                                           meta_database_name.c_str(),
-                                           hive_meta_database_name.c_str(),
-                                           config, elastic_index, elastic_batch_size,
-                                           elastic_issue_time, lru_cap);
-
-      reindexer->run();
+      if(reindex_of == "project") {
+        LOG_INFO("Create Elasticsearch index at " << elastic_index);
+        Reindexer *reindexer = new Reindexer(connection_string.c_str(),
+                                             database_name.c_str(),
+                                             meta_database_name.c_str(),
+                                             hive_meta_database_name.c_str(),
+                                             config, elastic_index, elastic_batch_size,
+                                             elastic_issue_time, lru_cap);
+        reindexer->run();
+      } else if(reindex_of == "featurestore") {
+        LOG_INFO("Create Elasticsearch index at " << elastic_featurestore_index);
+        FeaturestoreReindexer *reindexer = new FeaturestoreReindexer(connection_string.c_str(),
+                database_name.c_str(), meta_database_name.c_str(), hive_meta_database_name.c_str(), config,
+                elastic_featurestore_index, elastic_batch_size, elastic_issue_time, lru_cap);
+        reindexer->run();
+      }
```

Comment:
what about both? reindexing featurestore and project, maybe we should another condition for All?

Reply:
it is more complicated and adds little benefit... you could just first run project reindex and then featurestore reindex.

Reply:
fixed